browser/brave_rewards/rewards_tab_helper.cc

```diff
@@ -102,20 +104,26 @@ void RewardsTabHelper::DidFinishLoad(
   rewards_service_->OnLoad(tab_id_, validated_url);
 }
 
-void RewardsTabHelper::DidFinishNavigation(content::NavigationHandle* handle) {
-  if (!handle->IsInMainFrame() || !handle->HasCommitted() ||
-      handle->IsDownload()) {
+void RewardsTabHelper::DidFinishNavigation(
+    content::NavigationHandle* navigation_handle) {
+  if (!navigation_handle->HasCommitted() ||
+      !navigation_handle->IsInPrimaryMainFrame() || !rewards_service_) {
     return;
   }
 
-  auto id = GetPublisherIdFromURL(GetWebContents().GetLastCommittedURL());
-  SetPublisherIdForTab(id ? *id : "");
-
-  MaybeSavePublisherInfo();
+  auto* rfh = navigation_handle->GetRenderFrameHost();
 
-  if (rewards_service_) {
+  if (!navigation_handle->IsSameDocument()) {
```

Comment:
is it worth checking here to make sure that the URL is http(s)? otherwise it's not eliglble for rewards anyway

Reply:
Yes - I've added the check to `IsMediaPlatformURL`, which is used by `CreatorDetectionAgent` when determining whether to inject a script.